kahuna/public/js/services/label.js

```diff
@@ -34,44 +35,43 @@ labelService.factory('labelService',
     }
 
     function remove (image, label) {
-        var existingLabels = imageAccessor.readLabels(image);
-        var labelIndex = existingLabels.findIndex(lbl => lbl.data === label);
-        if (labelIndex !== -1) {
-            return existingLabels[labelIndex]
-                .delete()
-                .then(newLabels => apiPoll(() => untilLabelsEqual(image, newLabels.data)))
-                .then(newImage => {
-                    $rootScope.$emit('image-updated', newImage, image);
-                    return newImage;
-                });
-        }
-
-        // no-op
-        return Promise.resolve(image);
+      return batchRemove([image], label);
     }
 
     function add (image, labels) {
-        labels = labels.filter(label => label && label.trim().length > 0);
-
-        return image.data.userMetadata.data.labels
-            .post({data: labels})
-            .then(newLabels => apiPoll(() => untilLabelsEqual(image, newLabels.data)))
-            .then(newImage => {
-                $rootScope.$emit('image-updated', newImage, image);
-                return image;
-            });
+      return batchAdd([image], labels);
     }
 
-    function batchAdd (images, labels) {
-        return trackAll($rootScope, "label", images, image => add(image, labels));
+
+
+                        function batchAdd(images, labels) {
```

Comment:
Nit: Lots of indentation in this file

Reply:
prettier absolutely hates angular, happy to reflow